docs/part5/roofit.md

```diff
@@ -687,3 +687,110 @@ nominal_values = (MH=124.627 +/- 0.398094,resolution=1[C],norm_s=33.9097 +/- 11.
+An introductory presentation about likelihoods and interval estimation is available [here](https://indico.cern.ch/event/976099/contributions/4138517/).
+
+**Note: We will use python syntax in this section; you should use a .py script. Make sure to do `from ROOT import *` at the top of your script **
+
+We've seen how to create variables and pdfs, and how to fit a pdf to data. But what if we have a counting experiment, or a histogram template shape? And what about systematic uncertainties?  Let's build a likelihood
+for this:
+
+$\mathcal{L} \propto p(\text{data}|\text{parameters})$
+
+where our parameters are parameters of interest, $\mu$, and nuisance parameters, $\theta$. The nuisance parameters are constrained by external measurements, so we add constraint terms $\pi(\vec{\theta}_0|\vec{\theta})$
+
+So we have
+$\mathcal{L} \propto p(\text{data}|\mu,\vec{\theta})\cdot \pi(\vec{\theta}_0|\vec{\theta})$
+
+let's try to build the likelihood by hand for a 1-bin counting experiment.
+The data is the number of observed events $N$, and the probability is just a poisson probability $p(N|\lambda) = \frac{\lambda^N e^{-\lambda}}{N!}$, where $\lambda$ is the number of events expected in our signal+background model: $\lambda = \mu\cdot s(\vec{\theta}) + b(\vec{\theta})$. 
+
+In the expression, s and b are the numbers of expected signal- and background events, which both depend on the nuisance parameters. Let's start by building a simple likelihood function with one signal process and one background process. We'll assume there are no nuisance parameters for now. The number of observed events in data is 15, the expected number of signal events is 5 and the expected number of background events 8.1.
+
+It's easiest to use the RooFit workspace factory to build our model ([this tutorial](https://root.cern/doc/master/rf511__wsfactory__basic_8py.html) has more information on the factory syntax).
+
+```
+from ROOT import *
+w = RooWorkspace("w")
+```
+We need to create an expression for the number of events in our model, $\mu s +b$:
+
+```
+w.factory('expr::n("mu*s +b", mu[1.0,0,4], s[5],b[8.1])')
+```
+Now we can build the likelihood, which is just our poisson pdf:
+```
+w.factory('Poisson::poisN(N[15],n)')
+```
+
+To find the best-fit value for our parameter of interest $\mu$ we need to maximize the likelihood. In practice it's actually easier to minimize the **N**egative **l**og of the **l**ikelihood, or NLL:
+
+```
+w.factory('expr::NLL("-log(@0)",poisN)')
+```
+
+We can now use the RooMinimizer to find the minimum of the NLL
+
+
+```
+nll = w.function("NLL")
+minim = RooMinimizer(nll)
+minim.setErrorLevel(0.5)
+minim.minimize("Minuit2","migrad")
+bestfitnll = nll.getVal()
+```
+Notice that we need to set the error level to 0.5 to get the uncertainties (relying on Wilks' theorem!) - note that there is a more reliable way of extracting the confidence interval (explicitly rather than relying on migrad). We will discuss this a bit later in this section.

```

Comment:
Would this be `explicitly relying on hesse`? I don't know if the `minimize` call implicitly uses hesse but I imagine it does if the error is reported. Migrad can build an approximate hessian but idk if that is what is reported here.